self_balancing_tree.py

```diff
@@ -0,0 +1,128 @@
+# Hacker rank problem link: https://www.hackerrank.com/challenges/self-balancing-tree
+
+
+class Node:
+    def __init__(self,data , left=None, right=None):
+        self.data = data
+        self.height = 0
+
+
+def height(root):
+    if root:
+        return 1 + max(height(root.left), height(root.right))

```

Comment:
Do we need a recursive function knowing that nodes store the height information? If that's for the case where we need to actually recompute those values, the name should be updated to reflect this. This is important as the function is going to be O(height) vs O(1) for the case that's cached in the node.